python-checks/src/test/java/org/sonar/python/checks/AbstractDuplicateKeyCheckTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.python.checks;
+
+import com.sonar.sslr.api.AstNode;
+import java.math.BigDecimal;
+import java.util.function.Supplier;
+import org.junit.Test;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.ExpressionStatement;
+import org.sonar.plugins.python.api.tree.FileInput;
+import org.sonar.python.parser.PythonParser;
+import org.sonar.python.tree.PythonTreeMaker;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.sonar.python.checks.AbstractDuplicateKeyCheck.NumericValue;
+import static org.sonar.python.checks.AbstractDuplicateKeyCheck.StringValue;
+
+@SuppressWarnings("squid:S2701")
+public class AbstractDuplicateKeyCheckTest {
+
+    private static <A> void testHashEquals(
+      Supplier<A> firstColliding,
+      Supplier<A> secondColliding,
+      Supplier<A> unrelated,
```

Comment:
Is using suppliers necessary? I find their usage slightly harder to read and we calculate the first and second argument in every test anyways.